roles/cnf_cert/tasks/tests.yml

```diff
@@ -11,37 +23,85 @@
+    path: "{{ tnf_dir }}/result_files/cnf-certification-tests_junit.xml"
     state: absent
 
+# This way of launching CNF Cert Suite is deprecated since tnf v5.1.1.
 # The task automatically append these environment variables to tnf_env_vars:
 # - TNF_PARTNER_REPO
 # - TNF_IMAGE
 # - SUPPORT_IMAGE
-# We have to set ignore_errors to true because, if any tnf test fails, the
-# result of the execution will be failed.  We will check log files afterwards
-# to really determine if there were errors in the execution or not.
-- name: Run the CNF Test Suite
+# We have to set ignore_errors to true (for tnf < v5.0.4) because, if any tnf test
+# fails, the result of the execution will be failed. We will check log files
+# afterwards to really determine if there were errors in the execution or not.
+- name: Run the CNF Test Suite (tnf <= v5.1.0)
+  when:
+    - test_network_function_version is version("v5.1.0", "<=")
+    - test_network_function_version != "HEAD"
   vars:
     tnf_partner_repo: "{{ (dci_local_registry | length) | ternary(dci_local_registry, 'quay.io') }}/testnetworkfunction"
     support_image: "debug-partner:{{ support_image_version }}"
-    tnf_exec_ignore_errors: "{{ (test_network_function_version is version('v5.0.4', '>=') or test_network_function_version == 'HEAD') | ternary('false', 'true') }}"
+    tnf_exec_ignore_errors: "{{ (test_network_function_version is version('v5.0.4', '>=')) | ternary('false', 'true') }}"
   environment: "{{ tnf_env_vars | combine({'TNF_PARTNER_REPO': tnf_partner_repo, 'TNF_IMAGE': tnf_image, 'SUPPORT_IMAGE': support_image}) }}"
   ansible.builtin.shell: |
     set -x
 
-    ./run-tnf-container.sh -k {{ kubeconfig_path }} \
-    -t {{ tnf_dir }}/{{ test_network_function_project_name }} \
-    -o {{ tnf_dir }}/{{ test_network_function_project_name }} \
+    ./run-tnf-container.sh -k {{ tnf_dir }}/config_files/kubeconfig \
+    -t {{ tnf_dir }}/config_files \
+    -o {{ tnf_dir }}/result_files \
     {% if tnf_labels | length > 0 %}-l '{{ tnf_labels }}' \{% endif %}
     {% if tnf_dockercfg_path is defined %}-c {{ tnf_dockercfg_path }} \{% endif %}
-    -i {{ tnf_image }} &> {{ test_network_function_project_name }}/dci-tnf-execution.log
+    -i {{ tnf_image }} &> result_files/dci-tnf-execution.log
   args:
     chdir: "{{ tnf_dir }}"
   ignore_errors: "{{ tnf_exec_ignore_errors | bool }}"
 
+# For v5.1.1 in advance, keep using some environment variables to
+# keep the definition of tnf_env_vars, but would be deprecated at some point.
+- name: Run the CNF Test Suite (tnf >= v5.1.1 and HEAD)
+  when: test_network_function_version is version("v5.1.1", ">=") or test_network_function_version == "HEAD"
+  vars:
+    tnf_partner_repo: "{{ (dci_local_registry | length) | ternary(dci_local_registry, 'quay.io') }}/testnetworkfunction"
+    support_image: "debug-partner:{{ support_image_version }}"
+  environment: "{{ tnf_env_vars }}"
+  ansible.builtin.shell: |
```

Comment:
Why not use the podman module?

Reply:
Could be used, but since running this podman container requires a bunch of arguments and parameter, I preferred to use the shell module as a first step, similarly to what we do with [preflight](https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/roles/preflight/tasks/test_preflight_check_container_one_image.yml#L70-L96).
Let me finish the testing and I can try to explore that. Could be a good improvement here.

Reply:
@reihl , I tried here to use `podman` module: https://www.distributed-ci.io/jobs/2143a51b-307a-4fbe-8073-9f00b4b5c8ca/jobStates, but it's a bit tricky right now to do this transition. The execution doesn't like to work right now by using just exactly the same arguments I'm using with `shell`, and also, this would require a loop task to check the status of the container to confirm it's working fine.
I think it's better to keep this as it is right now, just to follow the trend that we already follow with preflight, as I said before, but since this role would require a refactor at some point, I'll take care of addressing that point in that moment. Is it fine for you?

Reply:
Yes, we tried and face problems that we can avoid when we using shell. I just want to put this modules in our consideration. 
Having said that, can we consider using set_fact and convert the container command to a variable?

Reply:
Sure, I'll have the usage of podman module into consideration when addressing the next refactor of this role. I think it will be benefitial, but as I said, as this new way is also a fix for a current problem, and the integration of podman module would require to add more complexity to the role, I'd prefer to postpone this.

Regarding using set_fact, sure, I'll define the command in a variable (mainly with `vars`, as it's only used in this task) and give a try.

Reply:
This is done, now the command is provided in a separate variable.